synapse/rest/client/v1/login.py

```diff
@@ -229,6 +240,15 @@ async def _do_other_login(self, login_submission: JsonDict) -> Dict[str, str]:
         else:
             qualified_user_id = UserID(identifier["user"], self.hs.hostname).to_string()
 
+        if login_submission["type"] == LoginRestServlet.APPSERVICE_TYPE:
```

Comment:
In accordance with above comment about multiple types, instead of one.

```suggestion
        if login_submission["type"] in LoginRestServlet.APPSERVICE_TYPES:
```

Reply:
Since we're not doing multiple types for now.., closing.